content: Add draft of the Source track. (#1037)
slsa-framework/slsa: docs/spec/v1.1/terminology.md

| Consumer | A party who uses software provided by a producer. The consumer may verify provenance for software they consume or delegate that responsibility to a separate verifier. | A developer who uses open source software distributions. A business that uses a point of sale system.
| Infrastructure provider | A party who provides software or services to other roles. | A package registry's maintainers. A build platform's maintainers.

> ### Source model

Reviewer: The build model introduces the space. Do you want to do something similar for the source model?

> ### Source model

Reviewer: What is the logic behind putting the source model here? Just that the source track would come before the build track?

Should we create a subheading for models and then alphabetize the various models within it?

Author: Alphabetizing SGTM.

We might want to reorganize this page before finalizing the next release. It's pretty hard to read.

The Source track is scoped to a single project that is controlled by some organization. That organization determines what Source level should apply to the project and administers technical controls to enforce that level.
| Term | Description
| --- | ---

> | Source | An identifiable set of text and binary files and associated metadata usually used as input for the build system (see SLSA Build Track).

Reviewer: No action. Just an observation... we mention binary files here explicitly. This has the potential to be relevant in any future dependency track that might require the dependencies to meet certain criteria (i.e. provenance).

> | Revision | The canonical source at a given point in time as identified by the version control system. Git identifies revisions by digest, but other version control systems may use other identifiers.

Reviewer: ```suggestion
| Revision | The canonical source at a given point in time as identified by the version control system. As an example, git identifies revisions by digest.
```
Do you want to be more specific about what that digest is?

Author: Done

I clarified that we mean tree hash, not commit hash. I think I noted that fact somewhere else in the PR, but I can't find it at the moment.

> | Change | A set of modifications to one or more source files and associated metadata. Change metadata MUST include any information required to situate the change in relation to other changes (e.g. parent Revision).

Reviewer: Would a word like locate be better than situate? Also `Revision` capitalization is inconsistent with later use.

Author: I think situate is the correct word. In my mind, locating doesn't include context but situate does. We want to know (at least) the parent revision, so we want some added context.

| Change History | A record of the history of changes that went into the revision.
| Version Control System | Software for tracking and managing changes to source. Git and Subversion are examples of version control systems.
| Version Control Platform | A service for hosting version controlled software. GitHub and GitLab are examples of version control platforms.

| Role | Description
| --- | ---
| Administrator | A human who can perform privileged operations on one or more projects. Privileged actions include, but are not limited to, modifying the change history and modifying project- or organization-wide security policies.
| Merger | The person who applies a change to the source. This person may be the submitter or a different trusted person, depending on the version control platform.
| Proposer | The human who proposes a particular change to the source.
| Reviewer | The human who reviews a particular proposed change to the source.
| Trusted person | A human who is authorized by the organization to propose and approve changes to the source.
| Trusted robot | A machine that is authorized by the organization to propose and/or approve changes to the source.

> | Untrusted person | A human who has limited access to the project. They MAY be able to read the source. They MAY be able to propose or review changes to the source. They MAY NOT approve changes to the source or perform any privileged actions on the project.

Reviewer: Can we merge any of these with the earlier `Roles` section or are these explicitly unique to source?

Author: I think this page needs a bit reformatting, or maybe we break it up into several pages. For now, I moved the existing Roles section to a subsection of the build model and kept these roles as a subsection of the source model.

    *checksum database* guarantees that all clients receive the same artifact
    for a given URL.

### Source model

The Source track is scoped to a single project that is controlled by some organization. That organization determines what Source level should apply to the project and administers technical controls to enforce that level.

> | Term | Description

Author: I reordered the terms conceptually.

| Change | A set of modifications to one or more source files and associated metadata. Change metadata MUST include any information required to situate the change in relation to other changes (e.g. parent revision).
| Change History | A record of the history of changes that went into the revision.
| Organization | A collection of people who collectively create the Source. Examples of organizations include an open-source projects, a company, or a team within a company.
| Revision | The canonical source at a given point in time as identified by the version control system. As an example, you can identify a git revision by its tree hash.
| Source | An identifiable set of text and binary files and associated metadata usually used as input for the build system (see SLSA Build Track).
| Source Control Platform | A service or suite of services for hosting version controlled software. GitHub and GitLab are examples of source control platforms, as are combinations of tools like Gerrit code reviews with GitHub source control.
| Version Control System | Software for tracking and managing changes to source. Git and Subversion are examples of version control systems.

#### Source Roles

> | Role | Description

Author: I did my best to reorder the roles conceptually, but it's a little tricky since none of the terms appear in other definitions. I'm open to other orderings.
